app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,14 @@
+class OrdersController < ApplicationController
+
+  def index
+
+    if @user = current_user
```

Comment:
Pretty much what I was thinking of, only thing might be to refactor it a bit to: 

```
if current_user
  @orders = current_user.orders
else
  #flash message and redirect
end
```
Otherwise I think it's ready to merge.